# КОММЕНТАРИЙ РЕВЬЮВЕРА
Привет Анастасия 👋!)

Меня зовут Сибирев Владислав, я осуществлял проверку проекта. Может перейдем сразу на <b>ты</b>? Если это неприемлемо, то обязательно дай знать)

Далее по ходу работы ты сможешь встретить мои комментарии, они будут отмечены разными цветами:

<div class="alert alert-block alert-success">
 ✔️ Похвала и возможности развития
</div>
<div class="alert alert-block alert-warning">
⚠️ Некритичные замечания, но стоит обратить внимание
</div>
<div class="alert alert-block alert-danger">
❌ Грубая ошибка, необходимо исправить
</div>


Заранее прошу не удалять мои комментарии, это важно. Если у тебя возникнут вопросы, обозначь их, я с радостью тебе помогу!)

Как вариант, можешь обозначить их так:
<div class="alert alert-block alert-info">
<b>КОММЕНТАРИЙ СТУДЕНТА:</b>
</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [47]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [ ]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [48]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [49]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [50]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [7]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [51]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [9]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [52]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [ ]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [53]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [54]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [12]:
data.duplicated().sum()

71

In [55]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [56]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [57]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [ ]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [58]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [59]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [60]:
def children_grouped(quantity):# Создание функции для категоризации семей по числу детей.
    try:
        if quantity == 0:
            return 'бездетные'
        elif 1 <= quantity <= 2:
            return 'малодетные'
        elif 3 <= quantity <= 5:
            return 'многодетные'
    except:
        pass

In [76]:
data_children_grouped = data.loc[:,['children','debt']]
data_children_grouped['children_category'] = data_children_grouped['children'].apply(children_grouped)
data_children_grouped = data_children_grouped.groupby('children_category').agg({'debt':['count','sum','mean']})
data_children_grouped = data_children_grouped.reset_index().droplevel(0, axis = 1)
data_children_grouped['mean'] = (data_children_grouped['mean']*100).round(2).astype('str') + '%'

In [78]:
data_children_grouped.style.hide_index()

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Давай здесь выведем полную таблицу, с количеством кредиторов, количество заемщиков)) Прописал ниже как лучше это сделать)

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь все отлично стало)) Все замечания исправила, таблица теперь отлично смотрится)) 

  </div>

**Вывод:** Клиенты, которые `не имеют детей`, чаще возвращают кредит в долг: доля должников составляет около `7,5%`. Клиенты, у которых `один или два ребенка`, хуже выплачивают кредит: доля невоврата составляет `9,3%`. Для анализа многодетных людей не хватает входных значений, чтобы установить закономерность.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Не бойся использовать в выводах цифры, также нужно сделать акцент на сбалансированности выборок. Примерно выводы будут выглядеть так: 
`люди, которые не имеют детей, лучше всего отдают кредит, доля должников здесь составляет с 7,5%, хуже обстоит дело с должниками, кто имеет одного или двух детей, здесь доля невозврата составляет соответственно 9,2% и 9,4%. Остальные категории уже некорректно сравнивать, так как в них очень маленькая выборка`)) <br>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Выводы переработала, молодец)) 

  </div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [6]:
def family_grouped(status):# Создание функции для группировки семейного положения клиентов
    if status == 0:
        return 'брак зарегестрирован'
    elif 2 <= status <= 3:
        return 'брак был зарегестрирован'
    else:
        return 'брак не зарегестрирован'

In [43]:
data_family_grouped = data.loc[:,['family_status','family_status_id','debt']]
data_family_grouped['marriage'] = data_family_grouped['family_status_id'].apply(family_grouped)
data_family_grouped = data_family_grouped.groupby(['marriage','family_status']).agg({'debt':['count','sum','mean']})
data_family_grouped = data_family_grouped.reset_index().droplevel(0, axis = 1)
data_family_grouped['mean'] = (data_family_grouped['mean']*100).round(2).astype('str') + '%'

In [46]:
data_family_grouped

count  sum   mean
0  брак был зарегестрирован              в разводе   1195   85  7.11%
1  брак был зарегестрирован         вдовец / вдова    960   63  6.56%
2      брак зарегестрирован        женат / замужем  12380  931  7.52%
3   брак не зарегестрирован  Не женат / не замужем   2813  274  9.74%
4   брак не зарегестрирован       гражданский брак   4177  388  9.29%

###### <div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

Здесь, молодец, логика правильная, ты получил все необходимые данные, которые нужно, но что можно оптимизировать (это касается изучение всех взаимосвязей):
1) Добавить с помощью функции  `mean`  долю должников, то есть вот это действие не понадобится `data_children_grouped['debt_conversion'] = data_children_grouped['debt']['sum']/data_children_grouped['debt']['count']` <br>
2) Столбец `mean` можно умножить на 100, это же проценты<br>
3) Для красоты лучше избавляться от мультииндексов, после сброса `reset_index()` поставь `droplevel(1, axis=1)`, названия столбцов сразу же станут красивыми;<br>

**Вывод:** Доля задолжноности `женатых/замужних` клиентов составляет `7,5%`, что ниже доли среди клиентов, состоящих в гражданском браке или `неженатых/незамужних`, для которых данное значение состовляет `9,2%` и `9,7%` соответсвенно. Необходимо отметить, что для анализа взята выборка из 12380 замужних людей, в то время как неженатых значительно меньше – 2813. Полученные значения конверсии долга помогают установить общую тенденцию среди анкетированных: люди, состоящие в зарегестрированном браке или вступашие в него в прошлом, вероятнее погасят кредит в установленный срок.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ В выводах давай будет использовать цифры, также  удели здесь внимание тоже сбалансированности выборок))

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь тоже все отлично по выводам))

  </div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [28]:
def income_grouped(income):# Создание функции для группировки клиентов по уровню дохода
    if 0 <= income < 50000:
        return 'низкий уровень дохода'
    elif 50000 <= income <= 200000:
        return 'средний уровень дохода'
    else:
        return 'высокий уровень дохода'

In [42]:
data_income_grouped = data.loc[:,['total_income','debt']]
data_income_grouped['income_class'] = data_income_grouped['total_income'].apply(income_grouped)
data_income_grouped = data_income_grouped.groupby('income_class').agg({'debt':['count','sum','mean']})
data_income_grouped = data_income_grouped.reset_index().droplevel(0, axis = 1)
data_income_grouped['mean'] = (data_income_grouped['mean']*100).round(2).astype('str') + '%'

In [41]:
data_income_grouped.style.hide_index()

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Здесь молодец, что укрупнила группы))

**Вывод:** Люди со `средним доходом` (предполагем, что это от 50000 до 200000) чаще задерживают выплату кредита: доля должников составляет `8,5%`. Лучше дела обстоят у людей с `высоким уровнем дохода` (свыше 200000): доля невыплат составляет `7,1%`. Нельзя оценить платежеспособность людей с низким доходом из-за малой выборки. В расчете также не приняты во внимание многие внешние факторы такие как семейное положение, количество детей или место проживания, отражающиеся на денежных возможностях клиента.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [39]:
data_purpose_grouped = data.groupby('purpose_category').agg({'debt':['count','sum','mean']})
data_purpose_grouped = data_purpose_grouped.reset_index().droplevel(0, axis = 1)
data_purpose_grouped['mean'] = (data_purpose_grouped['mean']*100).round(2).astype('str') + '%'

In [40]:
data_purpose_grouped.style.hide_index()

**Вывод:** Клиенты, оформившие кредит с целью `проведения свадьбы` или `покупки недвижимости`, чаще выплачивают кредит в установленный срок: доля задолжников составляет `7,9%` и `7,2%` соответственно. Доля задолжников среди 
клиентов, собирающихся проводить `операции с автомобилем` или `получать образование`, выше и составляет `9,3%` и `9,2%` соответсвенно. Более высокая платежеспособность среди клиентов, собирающихся приобрести жилье, что объясняется тем, что стоимость недвижимости гораздо выше остальных покупок, требует более стабильного заработка и взвешенного решения для преобретения. Клиент больше заинтересован в своевременных платежах, так как каждый месяц банк начисляет проценты на оставшийся долг,следовательно, сумма долга учеличивается. Например, в случае аннуитентных кредитов, сумма долга растет в геометрической прогрессии.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюера:</b> 

⚠️ Почему по твоему самые надежные заемщики те, кто берет ипотеку? Здесь можно порасуждать в выводах)) 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Самые большие суммы при ипотеке, поэтому банки предъявляют самые жесткие требования к ипотечникам, поэтому их проверяют вдоль и поперек при выдаче кредитов, плюс собственность находится в залоге у банка, не будешь платить она вернется к банку)) 
   
       
  </div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* В двух столбцах: **days_employed** и **total_income** отсутствует равное чисто пропущенных значений – `2174`. Вероятнее всего произошла техническая ошибка либо при формировании таблицы с исходными данными для анализа, либо при объединении данных в единый файл, либо при формировании исходной таблицы для анкетирования клиентов, либо по другой причине.

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Да, всё здраво. В принципе эти возможные причины сразу и приходят в голову. 
    
Однако, если пристально взглянуть на на наши данные, то увидим что пропуски в столбцах "трудовой стаж" и "ежемесячный доход" абсолютно идентичны. Количество ячеек с отсутствующими данными в обоих указанных столбцах совпадает и равно 2174. Маловероятно, что специалисты по выдаче кредитов, заполнив все без исключения ячейки других столбцов по каждому кредитополучателю, оставляли незаполненными именно два показателя, и сделали это ровно 2174 раза в каждом столбце. Даже не проводя проверку, во всех ли случаях незаполненные ячейки находятся в одной и то же строке с высокой долей вероятности можно утверждать, что произошла техническая ошибка при формировании таблицы с исходными данными (либо некий сбой во время выгрузки данных на анализ, либо при объединении данных в единый файл, который также должен был передаваться на анализ, либо эти данные не собирались).
</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медианные значения в отличие от средних значений дают более объективное представление о количественных переменных. Это объясняется тем, что при подсчете среднего арифметического мы берем максимальное и минимальное значения из представленных, при этом мы не учитываем повторяемость значений и их разнообразие. Это может привести к недостоверно завышенному заработку или к нереалистично низкому стажу работы. 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Медиана не подвержена выбросам, все верно)) 

### Шаг 4: общий вывод.

В ходе исследования было изучено влияние различных факторов на платежеспособность клиента. Были найдены следующие закономерности:
- клиенты, чье семейное положение связано с фактом возложения на себя **семейных уз**, с большей вероятностью погасят кредит в срок. Так, доля задолжноности **женатых/замужних** клиентов составляет **7,5%**, что **ниже примерно на 2%** доли неженатых/незамужних** клиентов; 
- кредиторы, взявшие кредит с целью проведения **операций с недвижимостью** или **проведения свадьбы**, также потенциально возращают кредит без отсрочек. Доля должников по сравнению с другими категориями **ниже на 2%** и составляет **7,2%** и **7,9%** соответственно;
- клиенты **без детей** закрывают чаще кредит без долга: доля должников составляет **7,5%**, что **ниже примерно на 1,5%** доли других категорий;
- клиенты из **среднего класса** (с зарабатной платой 50000-200000 рублей) чаще имеют задолжность: доля должников **8,5%**, что **примерно на 1,5% больше** доли других категорий.

Полученные результаты можно будет учитывать при построении модели кредитного скоринга.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюера:</b> 

⚠️ По структуре все отлично, но общий вывод - резюме этой работы. Большинство заказчиков посмотрят сначала его (а некоторые только его). Чтобы выводы не огорошивали заказчика своей очевидностью, стоит добавить в них цифр, всяких корреляций, процентов итд :) Сделай так, чтобы вся найденная в данных информация была отражена в выводе))

 <div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Выводы хорошо переработала, молодец)) Рекомендации даны банку, исследование проведено))
  
  
</div>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>
    
Анастасия, отлично поработала, видно что умеешь работать с кодом, в общем молодец, но нужно будет оптимизировать код при изучении взаимосвязей и поработать над выводами)) 
<br> Жду исправленного варианта)
  </div>

 <div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Анастасия, отличная работа, можно двигаться дальше)) Успехов в дальнейшем обучении)) 
  
  
</div>